parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/Tokens.scala

```diff
@@ -1,29 +1,67 @@
 package uk.gov.ons.addressIndex.parsers
 
-import uk.gov.ons.addressIndex.crfscala.CrfScala.{CrfToken, CrfTokenable, CrfTokens, Input}
+import com.typesafe.config.ConfigFactory
+import uk.gov.ons.addressIndex.crfscala.CrfScala.CrfTokenable
 
-//todo scaladoc this thing
+import scala.io.Source
+
+/**
+  * Tokenizer for the input
+  */
 object Tokens extends CrfTokenable {
 
-  type TokenIndicator = String
-  type Token = CrfToken
-  type Tokens = CrfTokens
+  private val config = ConfigFactory.load()
+
+  /**
+    * Tokenizes input into tokens (also removes counties, replaces synonyms)
+    * @param input the string to be tokenized
+    * @return Array of tokens
+    */
+  override def apply(input: String): Array[String] ={
+    val upperInput = input.toUpperCase()
+
+    val inputWithoutCounties = removeCounties(upperInput)
+
+    val tokens = inputWithoutCounties
+      .replace(" - ", "-")
+      .replace("- ", "-")
+      .replace(" -", "-")
```

Comment:
Sorry to be a pain, I know that I okay'ed the above, however, given todays work I think we need a slightly more complicated replace here. We want to remove the white spaces only if there are numbers around the hyphen, not if it's part of something else. In that case we actually want to remove it complete (my Python uses a regular expression to split the string before computing features).

I suggest we replace the above three replace lines with hyphens with something like this:
`\d+\s?-\s?\d+`
That matches with digits and optional white spaces. The replacement should be something like
`\d+-\d+`